audit/entry_formatter.go

```diff
@@ -82,15 +83,21 @@ func (f *EntryFormatter) Process(ctx context.Context, e *eventlogger.Event) (*ev
 		return nil, fmt.Errorf("%s: cannot parse event payload: %w", op, event.ErrInvalidParameter)
 	}
 
-	var result []byte
+	if a.Data == nil {
+		return nil, fmt.Errorf("%s: cannot audit event (%s) with no data: %w", op, a.Subtype, event.ErrInvalidParameter)
+	}
+
+	// Take a copy of the event data before we modify anything.
 	data := new(logical.LogInput)
```

Comment:
This memory allocation is unnecessary, since the **data** variable is later reassigned to point to the result of the copystructure.Copy function call.

I would replace it with:
```
var data *logical.LogInput
```